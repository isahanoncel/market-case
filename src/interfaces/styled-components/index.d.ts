// #region Global Imports
import 'styled-components';
// #endregion Global Imports

export type Icons = 'Bag' | 'ArrowLeft' | 'Check';

export type Colors =
  | 'primary'
  | 'gray'
  | 'secondary'
  | 'whiteBlue'
  | 'white'
  | 'background'
  | 'transparent'
  | 'innerBorder'
  | 'black100'
  | 'thinGray'
  | 'black500'
  | 'black600';

type Fonts = 'OpenSans';

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: Record<Colors, string>;
    fonts: Record<Fonts, string>;
  }
}
