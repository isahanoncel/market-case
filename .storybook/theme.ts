import { create } from '@storybook/theming';

export default create({
  base: 'light',
  brandTitle: 'Getir',
  brandUrl: 'https://getir.com',
  brandTarget: '_self',
});
