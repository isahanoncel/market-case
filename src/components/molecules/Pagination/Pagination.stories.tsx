import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import Pagination from '.';

export default {
  title: 'Molecules/Pagination',
  component: Pagination,
} as ComponentMeta<typeof Pagination>;

export const Default: ComponentStory<typeof Pagination> = (args) => (
  <Pagination {...args} />
);
