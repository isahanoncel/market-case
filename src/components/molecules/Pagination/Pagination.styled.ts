import styled from 'styled-components';
import StyledHelper from '../../../constants/styles/Helper';

export const PaginationWrapper = styled.div`
  ${StyledHelper.CenterContent}
  margin-top: 32px;
  padding-bottom: 100px;
`;

export const PrevNext = styled.div<{ disabled?: boolean }>`
  cursor: pointer;
  ${StyledHelper.CenterContent}
  color:${({ theme }) => theme.colors.primary};
  font-size: 14px;
  font-weight: 600;
  gap: 6px;
  ${(props) => props.disabled && 'opacity:0;'}
  &:first-of-type {
    margin-right: 56px;
  }
  &:last-of-type {
    margin-left: 56px;
    svg {
      transform: rotateY(180deg);
    }
  }
`;

export const PaginationItem = styled.div<{ isActive?: boolean }>`
  cursor: pointer;
  font-size: 14px;
  font-weight: 600;
  width: 32px;
  height: 40px;
  border-radius: 2px;
  ${StyledHelper.CenterContent}
  ${(props) =>
    props.isActive &&
    `color:${props.theme.colors.white};background-color:${props.theme.colors.primary};`}
`;
