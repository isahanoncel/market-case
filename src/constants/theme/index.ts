import { DefaultTheme } from 'styled-components';

export const COLORS = {
  white: '#fff',
  primary: '#1EA4CE',
  whiteBlue: '#F2F0FD',
  secondary: '#147594',
  black100: '#E0E0E0',
  black500: '#6F6F6F',
  black600: '#525252',
  gray: '#697488',
  background: '#E5E5E5',
  transparent: 'transparent',
  innerBorder: '#DFDEE2',
  thinGray: '#A8A8A8',
};

export const FONTS = {
  OpenSans: "'Open Sans', sans-serif",
};

const heroTheme: DefaultTheme = {
  colors: COLORS,
  fonts: FONTS,
};

export default heroTheme;
