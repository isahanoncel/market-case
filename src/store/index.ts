import { configureStore } from '@reduxjs/toolkit';
import { useDispatch, useSelector, TypedUseSelectorHook } from 'react-redux';
import products from './slices/products';
import basket from './slices/basket';

const store = configureStore({
  reducer: {
    products,
    basket,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

export default store;
