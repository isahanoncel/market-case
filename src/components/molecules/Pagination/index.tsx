import React, { FC, useCallback, useMemo } from 'react';
import Icon from '../../atoms/Icon';
import * as S from './Pagination.styled';

interface IPagination {
  page: number;
  setPage: (page: number) => void;
  itemLength: number;
  itemsPerPage: number;
}

const Pagination: FC<IPagination> = ({
  page,
  itemLength,
  itemsPerPage,
  setPage,
}) => {
  const pageCount = Math.ceil(itemLength / itemsPerPage);

  const handleChangePage = useCallback(
    (newPage: number) => {
      if (newPage >= 1 && newPage <= Math.ceil(itemLength / itemsPerPage))
        setPage(newPage);
    },
    [page],
  );
  const renderPagination = useMemo(() => {
    const pages = [];
    if (page > 1) {
      for (let i = page - 1; i < page; i += 1) {
        pages.push(i);
      }
    }

    if (page < pageCount - 1) {
      for (let i = page; i <= page + 2; i += 1) {
        pages.push(i);
      }
    } else {
      for (let i = page; i <= page + 1; i += 1) {
        if (i <= pageCount) {
          pages.push(i);
        }
      }
    }
    if (page < pageCount - 5) {
      pages.push(0);
      for (let i = pageCount - 3; i <= pageCount; i += 1) {
        pages.push(i);
      }
    }

    return (
      <>
        {pages.map((item) => {
          return item === 0 ? (
            <S.PaginationItem key={item} isActive={false}>
              ...
            </S.PaginationItem>
          ) : (
            <S.PaginationItem
              key={item}
              isActive={item === page}
              onClick={() => handleChangePage(item)}
            >
              {item}
            </S.PaginationItem>
          );
        })}
      </>
    );
  }, [page, itemLength]);

  const prevLink = useCallback(() => {
    return (
      <S.PrevNext
        disabled={page === 1}
        onClick={() => handleChangePage(page - 1)}
      >
        <Icon name="ArrowLeft" /> Prev
      </S.PrevNext>
    );
  }, [page]);

  const nextLink = useCallback(() => {
    if (page >= pageCount) {
      return null;
    }

    return (
      <S.PrevNext onClick={() => handleChangePage(page + 1)}>
        Next <Icon name="ArrowLeft" />
      </S.PrevNext>
    );
  }, [page]);

  if (itemsPerPage > itemLength) {
    return null;
  }
  return (
    <S.PaginationWrapper>
      {prevLink()}
      {renderPagination}
      {nextLink()}
    </S.PaginationWrapper>
  );
};

export default Pagination;
