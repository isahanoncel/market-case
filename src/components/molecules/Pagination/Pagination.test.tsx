import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import Pagination from '.';
import theme from '../../../constants/theme';
import store from '../../../store';

test('Pagination snapshot testing', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Pagination
            setPage={() => {}}
            itemLength={100}
            page={1}
            itemsPerPage={16}
          />
        </ThemeProvider>
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
