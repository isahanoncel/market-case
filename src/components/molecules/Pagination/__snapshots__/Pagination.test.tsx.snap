// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination snapshot testing 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 32px;
  padding-bottom: 100px;
}

.c1 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #1EA4CE;
  font-size: 14px;
  font-weight: 600;
  gap: 6px;
  opacity: 0;
}

.c1:first-of-type {
  margin-right: 56px;
}

.c1:last-of-type {
  margin-left: 56px;
}

.c1:last-of-type svg {
  -webkit-transform: rotateY(180deg);
  -ms-transform: rotateY(180deg);
  transform: rotateY(180deg);
}

.c4 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #1EA4CE;
  font-size: 14px;
  font-weight: 600;
  gap: 6px;
}

.c4:first-of-type {
  margin-right: 56px;
}

.c4:last-of-type {
  margin-left: 56px;
}

.c4:last-of-type svg {
  -webkit-transform: rotateY(180deg);
  -ms-transform: rotateY(180deg);
  transform: rotateY(180deg);
}

.c2 {
  cursor: pointer;
  font-size: 14px;
  font-weight: 600;
  width: 32px;
  height: 40px;
  border-radius: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  background-color: #1EA4CE;
}

.c3 {
  cursor: pointer;
  font-size: 14px;
  font-weight: 600;
  width: 32px;
  height: 40px;
  border-radius: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
>
  <div
    className="c1"
    disabled={true}
    onClick={[Function]}
  >
    <svg
      fill="#1EA4CE"
      height={16}
      viewBox="0 0 16 16"
      width={16}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M0.0667786 7.33281C0.110539 7.43791 0.171804 7.53425 0.259324 7.61308L6.38574 13.7438C6.72707 14.0854 7.27844 14.0854 7.61977 13.7438C7.9611 13.4023 7.9611 12.8505 7.61977 12.5089L2.98995 7.87582L13.1248 7.87582C13.6062 7.87582 14 7.4817 14 7C14 6.5183 13.6062 6.12418 13.1248 6.12418L2.9812 6.12418L7.61102 1.49109C7.95235 1.14951 7.95235 0.597748 7.61102 0.256178C7.26969 -0.0853924 6.71832 -0.0853923 6.37699 0.256178L0.250571 6.38693C0.171802 6.46575 0.110537 6.56209 0.0580247 6.66719C-0.020744 6.87739 -0.0207421 7.12262 0.0667786 7.33281Z"
        fill="#1EA4CE"
        fillRule="evenodd"
      />
    </svg>
     Prev
  </div>
  <div
    className="c2"
    onClick={[Function]}
  >
    1
  </div>
  <div
    className="c3"
    onClick={[Function]}
  >
    2
  </div>
  <div
    className="c3"
    onClick={[Function]}
  >
    3
  </div>
  <div
    className="c3"
  >
    ...
  </div>
  <div
    className="c3"
    onClick={[Function]}
  >
    4
  </div>
  <div
    className="c3"
    onClick={[Function]}
  >
    5
  </div>
  <div
    className="c3"
    onClick={[Function]}
  >
    6
  </div>
  <div
    className="c3"
    onClick={[Function]}
  >
    7
  </div>
  <div
    className="c4"
    onClick={[Function]}
  >
    Next 
    <svg
      fill="#1EA4CE"
      height={16}
      viewBox="0 0 16 16"
      width={16}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M0.0667786 7.33281C0.110539 7.43791 0.171804 7.53425 0.259324 7.61308L6.38574 13.7438C6.72707 14.0854 7.27844 14.0854 7.61977 13.7438C7.9611 13.4023 7.9611 12.8505 7.61977 12.5089L2.98995 7.87582L13.1248 7.87582C13.6062 7.87582 14 7.4817 14 7C14 6.5183 13.6062 6.12418 13.1248 6.12418L2.9812 6.12418L7.61102 1.49109C7.95235 1.14951 7.95235 0.597748 7.61102 0.256178C7.26969 -0.0853924 6.71832 -0.0853923 6.37699 0.256178L0.250571 6.38693C0.171802 6.46575 0.110537 6.56209 0.0580247 6.66719C-0.020744 6.87739 -0.0207421 7.12262 0.0667786 7.33281Z"
        fill="#1EA4CE"
        fillRule="evenodd"
      />
    </svg>
  </div>
</div>
`;
